xds/src/main/proto/centraldogma/xds/endpoint/v1/xds_endpoint.proto

```diff
@@ -0,0 +1,102 @@
+// Copyright 2024 LINE Corporation
+//
+// LINE Corporation licenses this file to you under the Apache License,
+// version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at:
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+syntax = "proto3";
+
+package centraldogma.xds.endpoint.v1;
+
+option java_multiple_files = true;
+option java_outer_classname = "XdsEndpointProto";
+option java_package = "com.linecorp.centraldogma.xds.endpoint.v1";
+
+import "envoy/config/endpoint/v3/endpoint.proto";
+
+import "google/api/annotations.proto";
+import "google/api/client.proto";
+import "google/api/field_behavior.proto";
+import "google/protobuf/empty.proto";
+
+// An XdsEndpointService provides methods to manage endpoints.
+service XdsEndpointService {
+
+  // Creates a new endpoint.
+  rpc CreateEndpoint(CreateEndpointRequest) returns (envoy.config.endpoint.v3.ClusterLoadAssignment) {
+    option (google.api.http) = {
+      post: "/api/v1/xds/{parent=groups/*}/endpoints"
+      body: "endpoint"
+    };
+  }
+
+  // Updates an existing endpoint.
+  rpc UpdateEndpoint(UpdateEndpointRequest) returns (envoy.config.endpoint.v3.ClusterLoadAssignment) {
+    option (google.api.http) = {
+      // ClusterLoadAssignment does not have the name field, so use endpoint_name.
+      patch: "/api/v1/xds/{endpoint_name=groups/*/endpoints/**}"
+      body: "endpoint"
+    };
+  }
+
+  // Deletes a endpoint.
+  rpc DeleteEndpoint(DeleteEndpointRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      delete: "/api/v1/xds/{name=groups/*/endpoints/**}"
+    };
+  }
+
+  // TODO(minwoox): Implement the following methods.
+  //rpc addLocalityLbEndpoint(addLocalityLbEndpointRequest) returns
```

Comment:
So do we consider `LocalityLbEndpoint` as a `resource`?  or do we use `UpdateEndpointRequest.update_mask`?

Reply:
This API will add a `LocalityLbEndpoint` to an existing `ClusterLoadAssignment`. I thought this would be useful for situations where each instance registers its information dynamically. The intended flow is:
- A user creates a `ClusterLoadAssignment` with empty endpoints.
- Each instance will (de)register its information via this API to the `ClusterLoadAssignment`.

Reply:
I see, I think we can worry about this API later then. I was wondering how this API fits into your idea of following the google AIP.

As a side-note, I also think an interesting point we will have to think about is how to guarantee consistency for concurrent `[add|remove]LocalityLbEndpoint` calls

Reply:
>  I was wondering how this API fits into your idea of following the google AIP.

I was trying to use a custom verb so we don't have to follow the create or update AIP
e.g. `patch: /api/v1/xds/{endpoint_name=groups/*/endpoints/**}:register"`

> we will have to think about is how to guarantee consistency for concurrent [add|remove]LocalityLbEndpoint calls

I was going to introduce a sort of CAS commit with a windowing. Will share the idea once it's concretized.